### DISCUSSION TITLE
httpx package guide?

### DISCUSSION THREAD

**Discussion by Xplo8E** [projectdiscovery/httpx#816] [Q&A] [answered]
is there any **documentation for using httpx as a package** in my tools, my tool collects subdomains with the help of subfinder package and stores them in an output file, bow I need to specify that subfinder **output file to the httpx package as an input file** to get hosts.

thanks

**Comment by forgedhallpass** [comment]
Have you read the README? There is a dedicated [section](https://github.com/projectdiscovery/httpx#using-httpx-as-a-library) on how to use `httpx` as a library.

**Comment by Xplo8E** [comment]
Yes, I know about it but my problem was, 
From the readme section 
It contains `inputfile` and `scanme.sh` I was confused about these two, from which one the input was gave to httpx. 
Can u please elaborate regarding this

**Comment by forgedhallpass** [comment, marked as answer]
You can take a look at the [`Options`](https://github.com/projectdiscovery/httpx/blob/master/runner/options.go#L133) struct and can set any of those values to be passed to the `runner`.

E.g. you can easily pass in a list of inputs using the following snippet:
```go
import "github.com/projectdiscovery/goflags"
// ...
options := runner.Options{
	Methods:         "GET",
	InputTargetHost: goflags.StringSlice{"scanme.sh", "projectdiscovery.io"},
}
```

**Comment by Xplo8E** [comment]
Thanks for helping
